src/lib/Liblog/log_event.c

```diff
@@ -122,3 +125,48 @@ log_event(int eventtype, int objclass, int sev, const char *objname, const char
 	if (will_log_event(eventtype))
 		log_record(eventtype, objclass, sev, objname, text);
 }
+
+/**
+ * @brief
+ * 	log_eventf - a combination of log_event() and printf()
+ * 
+ * @param[in] eventtype - event type
+ * @param[in] objclass - event object class 
+ * @param[in] sev - indication for whether to syslogging enabled or not
+ * @param[in] objname - object name stating log msg related to which object
+ * @param[in] fmt - format string
+ * @param[in] ... - arguments to format string
+ * 
+ * @return void
+ */
+void
+log_eventf(int eventtype, int objclass, int sev, const char *objname, const char *fmt, ...)
+{
+	va_list args;
+	int len;
+	char logbuf[LOG_BUF_SIZE];
+	char *buf;
+	
+	if (will_log_event(eventtype) == 0)
+		return;
+
+	va_start(args, fmt);
+
+	len = vsnprintf(logbuf, sizeof(logbuf), fmt, args);
+
+	if (len >= sizeof(logbuf)) {
+		buf = pbs_asprintf_format(len, fmt, args);
+		if (buf == NULL) {
+			va_end(args);
+			return;
```

Comment:
shouldn't we log something when this fails? Since this function returns a void, the user won't really know if anything is wrong.

Reply:
To what end?  The message would be very generic and not really helpful.  If you want a log message, I can put it in pbs_asprintf_format() when the malloc() fails, if you want.  Here all we'd log is 'log_eventf() failure' or something like that.  It wouldn't really be helpful.  We failed to format the string, so we would have no idea what log message had an error.

Reply:
The effect of this failing is that some log messages would potentially not get printed right? I think we should add some hint in the logs that this happened as the absence of log messages might get construed as "things look fine". So even a generic log message might help.

Reply:
Actually I'm ok with not doing this if you feel this is a 1 in a million scenario, so up to you

Reply:
The only failure cases are malloc() failure and if vsnprintf() returns an error.  Today neither are checked for.  This error is going to be very rare, so I don't think we should bother.